### PR TITLE
add sqlite3 support

### PR DESCRIPTION
**Pull request by yfblock**
1. Add sqlite3 support (memory version) and its related header
2. Modify the makefile of c_libax. It can use the makefile of the app folder now.

### CODE REVIEW

File: ulib/c_libax/src/string.c [hunk 1]
```diff
@@ -109,3 +109,48 @@ int strncmp(const char *_l, const char *_r, size_t n)
     for (; *l && *r && n && *l == *r; l++, r++, n--);
     return *l - *r;
 }
+
+size_t strcspn(const char *s1, const char *s2)
```
Comment (equation314) [inline ulib/c_libax/src/string.c:120]: Use bitmap to improve performance, refer to musl/src/string/strcspn.c

File: ulib/c_libax/include/sys/types.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+typedef int blkcnt_t;
+
+#define SEEK_SET 0
+#define SEEK_CUR 1
+#define SEEK_END 2
+
+#define _SC_PAGESIZE 30
+
+typedef unsigned int __off_t;
+// struct _IO_FILE
```
Comment (equation314) [inline ulib/c_libax/include/sys/types.h]: Remove the comments.

File: ulib/c_libax/include/sys/types.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * https://stackoverflow.com/questions/15976290/how-to-compare-nlink-t-to-int
+ */
+typedef unsigned int nlink_t;
+typedef int blksize_t;
+typedef int blkcnt_t;
+
+#define SEEK_SET     0
+#define SEEK_CUR     1
+#define SEEK_END     2
```
Comment (equation314) [inline ulib/c_libax/include/sys/types.h]: Move the above defines to `stdio.h`.

File: ulib/c_libax/include/sys/types.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+#define _SC_PAGESIZE 30
```
Comment (equation314) [inline ulib/c_libax/include/sys/types.h]: Move this define to `unistd.h`

File: ulib/c_libax/include/sys/ioctl.h
Comment (equation314) [inline ulib/c_libax/include/sys/ioctl.h:1]: Add `#ifdef __SYS_IOCTL_H__` ...

**Review by equation314 — CHANGES_REQUESTED** [review]

File: ulib/c_libax/src/string.c (hunk 1 above)
Comment (yfblock) [reply ulib/c_libax/src/string.c:120]: good point.

File: ulib/c_libax/Makefile [hunk 5]
```diff
@@ -79,10 +83,14 @@ $(obj_dir):
 	@echo
 	@echo Building C static library $(c_lib) ...
 	mkdir -p $@
+	mkdir -p $@/sys
```
Comment (equation314) [inline ulib/c_libax/Makefile]: Remove this.

File: ulib/c_libax/include/fcntl.h [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    __kernel_pid_t l_pid;
+#ifdef __ARCH_FLOCK_EXTRA_SYSID
+    __ARCH_FLOCK_EXTRA_SYSID
+#endif
+#ifdef __ARCH_FLOCK_PAD
+    __ARCH_FLOCK_PAD
+#endif
+};
+
+// struct flock64 {
```
Comment (equation314) [inline ulib/c_libax/include/fcntl.h]: Remove this.

File: ulib/c_libax/include/fcntl.h [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+ */
+#define F_OFD_GETLK  36
+#define F_OFD_SETLK  37
+#define F_OFD_SETLKW 38
+
+#define F_OWNER_TID  0
+#define F_OWNER_PID  1
+#define F_OWNER_PGRP 2
+
+// struct f_owner_ex {
```
Comment (equation314) [inline ulib/c_libax/include/fcntl.h]: Remove this or uncomment.

File: .clang-format [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
+Language: Cpp
+BasedOnStyle: LLVM
+IndentWidth: 4
+ColumnLimit: 100
+AllowShortBlocksOnASingleLine: Empty
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortLoopsOnASingleLine: true
+AllowShortIfStatementsOnASingleLine: Never
+AlignConsecutiveMacros: AcrossEmptyLines
```
Comment (equation314) [inline .clang-format]: Change to `AlignConsecutiveMacros: bool` and re-run `make fmt_c`.

File: Makefile [hunk 9]
```diff
@@ -144,10 +144,13 @@ clippy:
 fmt:
 	cargo fmt --all
 
+fmt_c:
+	clang-format --style=file:.clang-format -i $(shell find apps/c ulib/c_libax -iname *.c) $(shell find apps/c ulib/c_libax -iname *.h)
```
Comment (equation314) [inline Makefile]:
```
@clang-format --style=file -i $(shell find apps/c ulib/c_libax -iname *.c -o -iname *.h)
```

**Review by equation314 — APPROVED** [review]

**Pull request merged**